### DISCUSSION TITLE
Pin field

### DISCUSSION THREAD

**Discussion by pihcardozo** [Telefonica/mistica-design#1212] [Proposals]
### Goal

O intuito é provocar a criação de um novo componente. Um componente de OTP Input, que atualmente é utilizado para login do App Vivo, mas que não está componentizado. 

### Context

O componente OTP Input está se fazendo necessário neste momento, pois no Brasil a tendencia é que a validação de dois fatores seja cada vez mais comum, e o App Vivo está ingressando nesta tendencia.

### Impact

Já estamos utilizando o sistema OPT para logar no App Vivo, mas neste momento há fluxos internos que está tendo esta mesma necessidade, seja para cadastro de senha ou para validação de segurança em transação financeira e de pagamentos.   Isso traz mais segurança e credibilidade para os serviços prestados pela Vivo.

### Discussion Questions

Temos 3 necessidades para um mesmo layout de componente:

1 - Ter a função OTP com campos de 6 a 8 caracteres numéricos;

2 - Ter função de senha de 4 a 6 caracteres numéricos;

3 - Ter a função para cadastrar senha/PIN para acesso ao aplicativo com 4 caracteres numéricos com adição de biometria.

### Additional considerations

Por se tratar de um componente que está sendo requisitado para a Vivo, entendemos que há a necessidade de validação com mais empresas do Global. 

### Do you have a visual proposal?

https://www.figma.com/file/7c8RdB62AdlAgc8sogquRX/Password-Field-Alternative?node-id=0%3A1&t=3lXIGZdJ5pGmNcaO-1

**Comment by yceballost** [comment]
Hola Cardozo!

Este componente lo teníamos en el roadmap para algún momento, así que encantado de que esto forme parte de Mística!

Dejo por aquí la issue related https://github.com/Telefonica/mistica-design/issues/931

Os creamos branch en specs y podéis empezar a hacer la spec
https://www.figma.com/file/e8d3PkXitUJ1Jwgc9Y3mNc/branch/r2ZlMD7ctYzJ6XGYIcqfur/%F0%9F%94%B8-Input-Fields-%26-Forms-Component-Specs?type=design&node-id=0%3A1&t=m7yLtfSjts5yLPwx-1

**Comment by piresrodrigo** [comment]
@pihcardozo
Si no lo teníamos "componentizado" hasta ahora, sugiero seguir usándolo atómicamente, sin mistificarlo para o app actual. No quería poner energía en algo que no era una prioridad.

 @yceballost No sé si ves algún problema en seguir adelante con esa propuesta.

**Reply by yceballost** [reply]
No hay problema siempre y cuando el equipo que lo vaya a necesitar lo vaya a construir atómicamente
